src/DynamoCore/Scheduler/UpdateRenderPackageAsyncTask.cs

```diff
@@ -149,54 +149,122 @@ protected override void HandleTaskExecutionCore()
                 return;
             }
 
+            var package = factory.CreateRenderPackage();
+            package.DisplayLabels = displayLabels;
+            package.Description = tag;
+            package.IsSelected = isNodeSelected;
+
+            //Initialize the package here assuming we will only generate a single renderPackage for this node
+            //We set the AllowLegacyColorOperations so that we can catch tessellation implementations which 
+            //use the deprecated calls.  At that point we will roll back the changes to the renderPackage
+            //and call tessellate with a new renderPackage object with AllowLegacyColorOperations set to true.
+            if (package is IRenderPackageSupplement packageSupplement)
+            {
+                packageSupplement.AllowLegacyColorOperations = false;
+            }
+
+            GetRenderPackagesFromMirrorDataImp(outputPortId, mirrorData, package, tag);
+
+            if (package.MeshVertexColors.Any())
+            {
+                package.RequiresPerVertexColoration = true;
+            }
+
+            if (package.HasRenderingData)
+            {
+                renderPackageCache.Add(package, outputPortId);
+            }
+        }
+
+        private void GetRenderPackagesFromMirrorDataImp(
+            Guid outputPortId,
+            MirrorData mirrorData,
+            IRenderPackage package,
+            string labelKey)
+        {
+
             if (mirrorData.IsCollection)
             {
                 int count = 0;
                 foreach (var el in mirrorData.GetElements())
                 {
                     if (el.IsCollection || el.Data is IGraphicItem)
                     {
-                        string newTag = tag + ":" + count;
-                        GetRenderPackagesFromMirrorData(outputPortId, el, newTag, displayLabels);
+                        string newLabel = labelKey + ":" + count;
+                        GetRenderPackagesFromMirrorDataImp(outputPortId, el, package, newLabel);
                     }
-                    count = count + 1;
+                    count += 1;
                 }
             }
             else
             {
-                var graphicItem = mirrorData.Data as IGraphicItem;
-                if (graphicItem == null)
+                if (!(mirrorData.Data is IGraphicItem graphicItem))
                 {
                     return;
                 }
 
-                var package = factory.CreateRenderPackage();
-                var packageWithTransform = package as ITransformable;
-                package.Description = tag;
+                GetRenderPackageFromGraphicItem(outputPortId, graphicItem, package, labelKey);
+            }
+        }
+
+        private void GetRenderPackageFromGraphicItem(Guid outputPortId, IGraphicItem graphicItem, IRenderPackage package, string labelKey)
```

Comment:
Are you going to rename this to something more appropriate like you discussed earlier?